packages/lib/src/io-command.ts

```diff
@@ -73,10 +74,17 @@ export interface OutputOptions {
 	indentLevel: number
 }
 
+export type InputArgv = string[]
+
+export interface IdAndArgs<ID> {
+	id: ID
+	argv: InputArgv
+}
```

Comment:
I think it would be easier to read the code if we created a couple of extra specific-purpose "Id" types in the command classes instead (that would then have named properties like "capability" or "component").